File: backend/tests/test_oracle_bug.js
const fetch = require('node-fetch');

async function test() {
    const text = "I love the movie Inception because of its mind-bending plot.";
    console.log("Adding memory...");
    const resAdd = await fetch('http://127.0.0.1:5051/api/memory/ingest', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ text, timestamp: new Date().toISOString(), isSnippet: true })
    });
    console.log("Add response:", await resAdd.json());

    console.log("\nQuerying Oracle...");
    const resOracle = await fetch('http://127.0.0.1:5051/api/memory/recover', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ question: "What movies do I like?" })
    });
    const oracleJson = await resOracle.json();
    console.log("Oracle response:", JSON.stringify(oracleJson, null, 2));
}

test();

File: backend/tests/test_oracle_bug.py
import requests
import json
import time
from datetime import datetime, timezone

def test():
    text = "I love the movie Inception because of its mind-bending plot."
    print("Adding memory...")
    res_add = requests.post(
        'http://127.0.0.1:5051/api/memory/ingest',
        json={
            "text": text,
            "timestamp": datetime.now(timezone.utc).isoformat(),
            "isSnippet": True
        }
    )
    print("Add response:", res_add.json())

    time.sleep(2)

    print("\nQuerying Oracle...")
    res_oracle = requests.post(
        'http://127.0.0.1:5051/api/memory/recover',
        json={"question": "What movies do I like?"}
    )
    print("Oracle response:", json.dumps(res_oracle.json(), indent=2))

if __name__ == "__main__":
    test()
